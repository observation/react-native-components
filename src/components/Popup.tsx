import React from 'react'
import { Modal, StyleSheet, View } from 'react-native'

import { BlurView } from '@react-native-community/blur'

import useShowBlurView from '../hooks/useShowBlurView'
import theme from '../styles/theme'

type Props = {
  visible: boolean
  children?: React.ReactNode
}

const Popup = ({ children, visible }: Props) => {
  const showBlurView = useShowBlurView()

  return (
    <Modal transparent animationType="fade" visible={visible} style={{ flex: 1 }}>
      {showBlurView && <BlurView style={theme.absolute} blurType="light" blurAmount={2} />}
      <View style={styles.modalBackground}>
        <View style={styles.viewPort}>{children}</View>
      </View>
    </Modal>
  )
}

export default Popup

const styles = StyleSheet.create({
  modalBackground: {
    flex: 1,
    alignItems: 'stretch',
    justifyContent: 'center',
    flexDirection: 'column',
    backgroundColor: theme.color.modalBackground,
  },
  viewPort: {
    flexDirection: 'column',
    justifyContent: 'center',
    backgroundColor: 'transparent',
  },
})
