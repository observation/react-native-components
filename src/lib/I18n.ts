import { I18nManager } from 'react-native'

import Countries from 'i18n-iso-countries'
import { I18n as I18nJs } from 'i18n-js'
import 'moment/min/locales'
import { Scope, TranslateOptions } from 'i18n-js/typings/typing'
import moment from 'moment'
import * as RNLocalize from 'react-native-localize'

import Log from './Log'

Countries.registerLocale(require('i18n-iso-countries/langs/en.json'))
Countries.registerLocale(require('i18n-iso-countries/langs/nl.json'))
Countries.registerLocale(require('i18n-iso-countries/langs/de.json'))
Countries.registerLocale(require('i18n-iso-countries/langs/es.json'))
Countries.registerLocale(require('i18n-iso-countries/langs/fr.json'))

interface SupportedLanguages {
  en: Object
  nl: Object
  de: Object
  fr: Object
  es: Object
}

interface Api {
  setLanguage(language: string): void
}

/** Wrapper class around i18n-js, adding a few more methods */
class I18n {
  i18nJs = new I18nJs()

  constructor(private translations: SupportedLanguages, private api: Api) {
    this.i18nJs.translations = translations
    this.i18nJs.enableFallback = false
    this.configure()
  }

  setLocale = (locale: string) => {
    Log.debug('i18n:setLocale', locale)
    this.i18nJs.locale = locale
    moment.locale(locale)
    this.api.setLanguage(locale)
  }

  get locale() {
    return this.i18nJs.locale
  }

  /** Handle setting an initial or changed locale */
  configure = () => {
    const fallback = { languageTag: 'en', isRTL: false }
    const languages = Object.keys(this.translations)
    const { languageTag, isRTL } = RNLocalize.findBestAvailableLanguage(languages) || fallback
    Log.debug('i18n:configure', languageTag)
    I18nManager.forceRTL(isRTL) // optional, you might not want to handle RTL
    this.setLocale(languageTag)
  }

  getCountryFromLocale = () => {
    Log.debug('i18n:getCountryFromLocale')

    const fallback = 'NL'
    const locales = RNLocalize.getLocales()
    return locales.length > 0 ? locales[0].countryCode : fallback
  }

  getLanguageFromLocale = () => {
    Log.debug('i18n:getLanguageFromLocale')

    const locales = RNLocalize.getLocales()
    return locales.length > 0 ? locales[0].languageCode : null
  }

  country = (code: string) => Countries.getName(code, this.locale, { select: 'official' })

  t = (key: Scope, options?: TranslateOptions | undefined) => this.i18nJs.t(key, options)
}

export default I18n
export type { Api }
