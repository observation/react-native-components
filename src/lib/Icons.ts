import { IconDefinition, IconName as FontawesomeIconName } from '@fortawesome/fontawesome-svg-core'
import { faAlignJustify as faAlignJustifyLight } from '@fortawesome/pro-light-svg-icons/faAlignJustify'
import { faAnalytics as faAnalyticsLight } from '@fortawesome/pro-light-svg-icons/faAnalytics'
import { faAngleDown as faAngleDownLight } from '@fortawesome/pro-light-svg-icons/faAngleDown'
import { faAngleUp as faAngleUpLight } from '@fortawesome/pro-light-svg-icons/faAngleUp'
import { faArrowCircleRight as faArrowCircleRightLight } from '@fortawesome/pro-light-svg-icons/faArrowCircleRight'
import { faArrowsRotate as faArrowsRotateLight } from '@fortawesome/pro-light-svg-icons/faArrowsRotate'
import { faArrowUpFromBracket as faArrowUpFromBracketLight } from '@fortawesome/pro-light-svg-icons/faArrowUpFromBracket'
import { faArrowUpFromSquare as faArrowUpFromSquareLight } from '@fortawesome/pro-light-svg-icons/faArrowUpFromSquare'
import { faBadge as faBadgeLight } from '@fortawesome/pro-light-svg-icons/faBadge'
import { faBadgeCheck as faBadgeCheckLight } from '@fortawesome/pro-light-svg-icons/faBadgeCheck'
import { faBanBug as faBanBugLight } from '@fortawesome/pro-light-svg-icons/faBanBug'
import { faCalendarDay as faCalendarDayLight } from '@fortawesome/pro-light-svg-icons/faCalendarDay'
import { faCalendarRange as faCalendarRangeLight } from '@fortawesome/pro-light-svg-icons/faCalendarRange'
import { faCamera as faCameraLight } from '@fortawesome/pro-light-svg-icons/faCamera'
import { faCertificate as faCertificateLight } from '@fortawesome/pro-light-svg-icons/faCertificate'
import { faChartColumn as faChartColumnLight } from '@fortawesome/pro-light-svg-icons/faChartColumn'
import { faCheck as faCheckLight } from '@fortawesome/pro-light-svg-icons/faCheck'
import { faCheckCircle as faCheckCircleLight } from '@fortawesome/pro-light-svg-icons/faCheckCircle'
import { faCheckSquare as faCheckSquareLight } from '@fortawesome/pro-light-svg-icons/faCheckSquare'
import { faChevronCircleLeft as faChevronCircleLeftLight } from '@fortawesome/pro-light-svg-icons/faChevronCircleLeft'
import { faChevronCircleRight as faChevronCircleRightLight } from '@fortawesome/pro-light-svg-icons/faChevronCircleRight'
import { faChevronLeft as faChevronLeftLight } from '@fortawesome/pro-light-svg-icons/faChevronLeft'
import { faChevronRight as faChevronRightLight } from '@fortawesome/pro-light-svg-icons/faChevronRight'
import { faCircle as faCircleLight } from '@fortawesome/pro-light-svg-icons/faCircle'
import { faCircleCheck as faCircleCheckLight } from '@fortawesome/pro-light-svg-icons/faCircleCheck'
import { faCircleExclamation as faCircleExclamationLight } from '@fortawesome/pro-light-svg-icons/faCircleExclamation'
import { faCirclePlus as faCirclePlusLight } from '@fortawesome/pro-light-svg-icons/faCirclePlus'
import { faCircleStop as faCircleStopLight } from '@fortawesome/pro-light-svg-icons/faCircleStop'
import { faCircleUser as faCircleUserLight } from '@fortawesome/pro-light-svg-icons/faCircleUser'
import { faCircleXmark as faCircleXmarkLight } from '@fortawesome/pro-light-svg-icons/faCircleXmark'
import { faClone as faCloneLight } from '@fortawesome/pro-light-svg-icons/faClone'
import { faCloudArrowDown as faCloudArrowDownLight } from '@fortawesome/pro-light-svg-icons/faCloudArrowDown'
import { faCog as faCogLight } from '@fortawesome/pro-light-svg-icons/faCog'
import { faCommentAltExclamation as faCommentAltExclamationLight } from '@fortawesome/pro-light-svg-icons/faCommentAltExclamation'
import { faCropAlt as faCropAltLight } from '@fortawesome/pro-light-svg-icons/faCropAlt'
import { faDash as faDashLight } from '@fortawesome/pro-light-svg-icons/faDash'
import { faEnvelope as faEnvelopeLight } from '@fortawesome/pro-light-svg-icons/faEnvelope'
import { faExclamationTriangle as faExclamationTriangleLight } from '@fortawesome/pro-light-svg-icons/faExclamationTriangle'
import { faExternalLink as faExternalLinkLight } from '@fortawesome/pro-light-svg-icons/faExternalLink'
import { faEye as faEyeLight } from '@fortawesome/pro-light-svg-icons/faEye'
import { faEyeSlash as faEyeSlashLight } from '@fortawesome/pro-light-svg-icons/faEyeSlash'
import { faFileLines as faFileLinesLight } from '@fortawesome/pro-light-svg-icons/faFileLines'
import { faGlobeEurope as faGlobeEuropeLight } from '@fortawesome/pro-light-svg-icons/faGlobeEurope'
import { faHeart as faHeartLight } from '@fortawesome/pro-light-svg-icons/faHeart'
import { faHexagon as faHexagonLight } from '@fortawesome/pro-light-svg-icons/faHexagon'
import { faImages as faImagesLight } from '@fortawesome/pro-light-svg-icons/faImages'
import { faInfoCircle as faInfoCircleLight } from '@fortawesome/pro-light-svg-icons/faInfoCircle'
import { faLanguage as faLanguageLight } from '@fortawesome/pro-light-svg-icons/faLanguage'
import { faLocationCheck as faLocationCheckLight } from '@fortawesome/pro-light-svg-icons/faLocationCheck'
import { faLocationCrosshairs as faLocationCrosshairsLight } from '@fortawesome/pro-light-svg-icons/faLocationCrosshairs'
import { faMapMarkerAlt as faMapMarkerAltLight } from '@fortawesome/pro-light-svg-icons/faMapMarkerAlt'
import { faMapMarkerAltSlash as faMapMarkerAltSlashLight } from '@fortawesome/pro-light-svg-icons/faMapMarkerAltSlash'
import { faMars as faMarsLight } from '@fortawesome/pro-light-svg-icons/faMars'
import { faMinus as faMinusLight } from '@fortawesome/pro-light-svg-icons/faMinus'
import { faPaperPlaneTop as faPaperPlaneToplight } from '@fortawesome/pro-light-svg-icons/faPaperPlaneTop'
import { faPenToSquare as faPenToSquareLight } from '@fortawesome/pro-light-svg-icons/faPenToSquare'
import { faPlayCircle as faPlayCircleLight } from '@fortawesome/pro-light-svg-icons/faPlayCircle'
import { faPlus as faPlusLight } from '@fortawesome/pro-light-svg-icons/faPlus'
import { faQrcode as faQrcodeLight } from '@fortawesome/pro-light-svg-icons/faQrcode'
import { faQuestionSquare as faQuestionSquareLight } from '@fortawesome/pro-light-svg-icons/faQuestionSquare'
import { faRandom as faRandomLight } from '@fortawesome/pro-light-svg-icons/faRandom'
import { faRectangleHistory as faRectangleHistoryLight } from '@fortawesome/pro-light-svg-icons/faRectangleHistory'
import { faRightToBracket as faRightToBracketLight } from '@fortawesome/pro-light-svg-icons/faRightToBracket'
import { faRoute as faRouteLight } from '@fortawesome/pro-light-svg-icons/faRoute'
import { faShareNodes as faShareNodesLight } from '@fortawesome/pro-light-svg-icons/faShareNodes'
import { faShield as faShieldLight } from '@fortawesome/pro-light-svg-icons/faShield'
import { faShieldCheck as faShieldCheckLight } from '@fortawesome/pro-light-svg-icons/faShieldCheck'
import { faSlidersH as faSlidersHLight } from '@fortawesome/pro-light-svg-icons/faSlidersH'
import { faSquare as faSquareLight } from '@fortawesome/pro-light-svg-icons/faSquare'
import { faStars as faStarsLight } from '@fortawesome/pro-light-svg-icons/faStars'
import { faSwatchbook as faSwatchbookLight } from '@fortawesome/pro-light-svg-icons/faSwatchbook'
import { faTally as faTallyLight } from '@fortawesome/pro-light-svg-icons/faTally'
import { faTimes as faTimesLight } from '@fortawesome/pro-light-svg-icons/faTimes'
import { faTrashAlt as faTrashAltLight } from '@fortawesome/pro-light-svg-icons/faTrashAlt'
import { faTriangle as faTriangleLight } from '@fortawesome/pro-light-svg-icons/faTriangle'
import { faUndo as faUndoLight } from '@fortawesome/pro-light-svg-icons/faUndo'
import { faUser as faUserLight } from '@fortawesome/pro-light-svg-icons/faUser'
import { faUserCheck as faUserCheckLight } from '@fortawesome/pro-light-svg-icons/faUserCheck'
import { faUserGroup as faUserGroupLight } from '@fortawesome/pro-light-svg-icons/faUserGroup'
import { faUserPlus as faUserPlusLight } from '@fortawesome/pro-light-svg-icons/faUserPlus'
import { faUserSlash as faUserSlashLight } from '@fortawesome/pro-light-svg-icons/faUserSlash'
import { faUsersSlash as faUsersSlashLight } from '@fortawesome/pro-light-svg-icons/faUsersSlash'
import { faVenus as faVenusLight } from '@fortawesome/pro-light-svg-icons/faVenus'
import { faAlignJustify as faAlignJustifySolid } from '@fortawesome/pro-solid-svg-icons/faAlignJustify'
import { faAnalytics as faAnalyticsSolid } from '@fortawesome/pro-solid-svg-icons/faAnalytics'
import { faAngleDown as faAngleDownSolid } from '@fortawesome/pro-solid-svg-icons/faAngleDown'
import { faAngleUp as faAngleUpSolid } from '@fortawesome/pro-solid-svg-icons/faAngleUp'
import { faArrowCircleRight as faArrowCircleRightSolid } from '@fortawesome/pro-solid-svg-icons/faArrowCircleRight'
import { faArrowsRotate as faArrowsRotateSolid } from '@fortawesome/pro-solid-svg-icons/faArrowsRotate'
import { faArrowUpFromBracket as faArrowUpFromBracketSolid } from '@fortawesome/pro-solid-svg-icons/faArrowUpFromBracket'
import { faArrowUpFromSquare as faArrowUpFromSquareSolid } from '@fortawesome/pro-solid-svg-icons/faArrowUpFromSquare'
import { faBadge as faBadgeSolid } from '@fortawesome/pro-solid-svg-icons/faBadge'
import { faBadgeCheck as faBadgeCheckSolid } from '@fortawesome/pro-solid-svg-icons/faBadgeCheck'
import { faBanBug as faBanBugSolid } from '@fortawesome/pro-solid-svg-icons/faBanBug'
import { faCalendarDay as faCalendarDaySolid } from '@fortawesome/pro-solid-svg-icons/faCalendarDay'
import { faCalendarRange as faCalendarRangeSolid } from '@fortawesome/pro-solid-svg-icons/faCalendarRange'
import { faCamera as faCameraSolid } from '@fortawesome/pro-solid-svg-icons/faCamera'
import { faCertificate as faCertificateSolid } from '@fortawesome/pro-solid-svg-icons/faCertificate'
import { faChartColumn as faChartColumnSolid } from '@fortawesome/pro-solid-svg-icons/faChartColumn'
import { faCheck as faCheckSolid } from '@fortawesome/pro-solid-svg-icons/faCheck'
import { faCheckCircle as faCheckCircleSolid } from '@fortawesome/pro-solid-svg-icons/faCheckCircle'
import { faCheckSquare as faCheckSquareSolid } from '@fortawesome/pro-solid-svg-icons/faCheckSquare'
import { faChevronCircleLeft as faChevronCircleLeftSolid } from '@fortawesome/pro-solid-svg-icons/faChevronCircleLeft'
import { faChevronCircleRight as faChevronCircleRightSolid } from '@fortawesome/pro-solid-svg-icons/faChevronCircleRight'
import { faChevronLeft as faChevronLeftSolid } from '@fortawesome/pro-solid-svg-icons/faChevronLeft'
import { faChevronRight as faChevronRightSolid } from '@fortawesome/pro-solid-svg-icons/faChevronRight'
import { faCircle as faCircleSolid } from '@fortawesome/pro-solid-svg-icons/faCircle'
import { faCircleCheck as faCircleCheckSolid } from '@fortawesome/pro-solid-svg-icons/faCircleCheck'
import { faCircleExclamation as faCircleExclamationSolid } from '@fortawesome/pro-solid-svg-icons/faCircleExclamation'
import { faCirclePlus as faCirclePlusSolid } from '@fortawesome/pro-solid-svg-icons/faCirclePlus'
import { faCircleStop as faCircleStopSolid } from '@fortawesome/pro-solid-svg-icons/faCircleStop'
import { faCircleUser as faCircleUserSolid } from '@fortawesome/pro-solid-svg-icons/faCircleUser'
import { faCircleXmark as faCircleXmarkSolid } from '@fortawesome/pro-solid-svg-icons/faCircleXmark'
import { faClone as faCloneSolid } from '@fortawesome/pro-solid-svg-icons/faClone'
import { faCloudArrowDown as faCloudArrowDownSolid } from '@fortawesome/pro-solid-svg-icons/faCloudArrowDown'
import { faCog as faCogSolid } from '@fortawesome/pro-solid-svg-icons/faCog'
import { faCommentAltExclamation as faCommentAltExclamationSolid } from '@fortawesome/pro-solid-svg-icons/faCommentAltExclamation'
import { faCropAlt as faCropAltSolid } from '@fortawesome/pro-solid-svg-icons/faCropAlt'
import { faDash as faDashSolid } from '@fortawesome/pro-solid-svg-icons/faDash'
import { faEnvelope as faEnvelopeSolid } from '@fortawesome/pro-solid-svg-icons/faEnvelope'
import { faExclamationTriangle as faExclamationTriangleSolid } from '@fortawesome/pro-solid-svg-icons/faExclamationTriangle'
import { faExternalLink as faExternalLinkSolid } from '@fortawesome/pro-solid-svg-icons/faExternalLink'
import { faEye as faEyeSolid } from '@fortawesome/pro-solid-svg-icons/faEye'
import { faEyeSlash as faEyeSlashSolid } from '@fortawesome/pro-solid-svg-icons/faEyeSlash'
import { faFileLines as faFileLinesSolid } from '@fortawesome/pro-solid-svg-icons/faFileLines'
import { faGlobeEurope as faGlobeEuropeSolid } from '@fortawesome/pro-solid-svg-icons/faGlobeEurope'
import { faHeart as faHeartSolid } from '@fortawesome/pro-solid-svg-icons/faHeart'
import { faHexagon as faHexagonSolid } from '@fortawesome/pro-solid-svg-icons/faHexagon'
import { faImages as faImagesSolid } from '@fortawesome/pro-solid-svg-icons/faImages'
import { faInfoCircle as faInfoCircleSolid } from '@fortawesome/pro-solid-svg-icons/faInfoCircle'
import { faLanguage as faLanguageSolid } from '@fortawesome/pro-solid-svg-icons/faLanguage'
import { faLocationCheck as faLocationCheckSolid } from '@fortawesome/pro-solid-svg-icons/faLocationCheck'
import { faLocationCrosshairs as faLocationCrosshairsSolid } from '@fortawesome/pro-solid-svg-icons/faLocationCrosshairs'
import { faMapMarkerAlt as faMapMarkerAltSolid } from '@fortawesome/pro-solid-svg-icons/faMapMarkerAlt'
import { faMapMarkerAltSlash as faMapMarkerAltSlashSolid } from '@fortawesome/pro-solid-svg-icons/faMapMarkerAltSlash'
import { faMars as faMarsSolid } from '@fortawesome/pro-solid-svg-icons/faMars'
import { faMinus as faMinusSolid } from '@fortawesome/pro-solid-svg-icons/faMinus'
import { faPaperPlaneTop as faPaperPlaneTopSolid } from '@fortawesome/pro-solid-svg-icons/faPaperPlaneTop'
import { faPenToSquare as faPenToSquareSolid } from '@fortawesome/pro-solid-svg-icons/faPenToSquare'
import { faPlayCircle as faPlayCircleSolid } from '@fortawesome/pro-solid-svg-icons/faPlayCircle'
import { faPlus as faPlusSolid } from '@fortawesome/pro-solid-svg-icons/faPlus'
import { faQrcode as faQrcodeSolid } from '@fortawesome/pro-solid-svg-icons/faQrcode'
import { faQuestionSquare as faQuestionSquareSolid } from '@fortawesome/pro-solid-svg-icons/faQuestionSquare'
import { faRandom as faRandomSolid } from '@fortawesome/pro-solid-svg-icons/faRandom'
import { faRectangleHistory as faRectangleHistorySolid } from '@fortawesome/pro-solid-svg-icons/faRectangleHistory'
import { faRightToBracket as faRightToBracketSolid } from '@fortawesome/pro-solid-svg-icons/faRightToBracket'
import { faRoute as faRouteSolid } from '@fortawesome/pro-solid-svg-icons/faRoute'
import { faShareNodes as faShareNodesSolid } from '@fortawesome/pro-solid-svg-icons/faShareNodes'
import { faShield as faShieldSolid } from '@fortawesome/pro-solid-svg-icons/faShield'
import { faShieldCheck as faShieldCheckSolid } from '@fortawesome/pro-solid-svg-icons/faShieldCheck'
import { faSlidersH as faSlidersHSolid } from '@fortawesome/pro-solid-svg-icons/faSlidersH'
import { faSquare as faSquareSolid } from '@fortawesome/pro-solid-svg-icons/faSquare'
import { faStars as faStarsSolid } from '@fortawesome/pro-solid-svg-icons/faStars'
import { faSwatchbook as faSwatchbookSolid } from '@fortawesome/pro-solid-svg-icons/faSwatchbook'
import { faTally as faTallySolid } from '@fortawesome/pro-solid-svg-icons/faTally'
import { faTimes as faTimesSolid } from '@fortawesome/pro-solid-svg-icons/faTimes'
import { faTrashAlt as faTrashAltSolid } from '@fortawesome/pro-solid-svg-icons/faTrashAlt'
import { faTriangle as faTriangleSolid } from '@fortawesome/pro-solid-svg-icons/faTriangle'
import { faUndo as faUndoSolid } from '@fortawesome/pro-solid-svg-icons/faUndo'
import { faUser as faUserSolid } from '@fortawesome/pro-solid-svg-icons/faUser'
import { faUserCheck as faUserCheckSolid } from '@fortawesome/pro-solid-svg-icons/faUserCheck'
import { faUserGroup as faUserGroupSolid } from '@fortawesome/pro-solid-svg-icons/faUserGroup'
import { faUserPlus as faUserPlusSolid } from '@fortawesome/pro-solid-svg-icons/faUserPlus'
import { faUserSlash as faUserSlashSolid } from '@fortawesome/pro-solid-svg-icons/faUserSlash'
import { faUsersSlash as faUsersSlashSolid } from '@fortawesome/pro-solid-svg-icons/faUsersSlash'
import { faVenus as faVenusSolid } from '@fortawesome/pro-solid-svg-icons/faVenus'

type IconName = Extract<
  FontawesomeIconName,
  | 'align-justify'
  | 'analytics'
  | 'angle-down'
  | 'angle-up'
  | 'arrow-circle-right'
  | 'arrow-up-from-bracket'
  | 'arrow-up-from-square'
  | 'arrows-rotate'
  | 'badge-check'
  | 'badge'
  | 'ban-bug'
  | 'calendar-day'
  | 'calendar-range'
  | 'camera'
  | 'certificate'
  | 'chart-column'
  | 'check-circle'
  | 'check-square'
  | 'check'
  | 'chevron-circle-left'
  | 'chevron-circle-right'
  | 'chevron-left'
  | 'chevron-right'
  | 'circle-check'
  | 'circle-exclamation'
  | 'circle-plus'
  | 'circle-stop'
  | 'circle-user'
  | 'circle-xmark'
  | 'circle'
  | 'clone'
  | 'cloud-arrow-down'
  | 'cog'
  | 'comment-alt-exclamation'
  | 'crop-alt'
  | 'dash'
  | 'envelope'
  | 'exclamation-triangle'
  | 'external-link'
  | 'eye-slash'
  | 'eye'
  | 'file-lines'
  | 'globe-europe'
  | 'heart'
  | 'hexagon'
  | 'images'
  | 'info-circle'
  | 'language'
  | 'location-check'
  | 'location-crosshairs'
  | 'map-marker-alt-slash'
  | 'map-marker-alt'
  | 'mars'
  | 'minus'
  | 'paper-plane-top'
  | 'pen-to-square'
  | 'play-circle'
  | 'plus'
  | 'qrcode'
  | 'question-square'
  | 'random'
  | 'rectangle-history'
  | 'right-to-bracket'
  | 'route'
  | 'share-nodes'
  | 'shield'
  | 'shield-check'
  | 'sliders-h'
  | 'square'
  | 'stars'
  | 'swatchbook'
  | 'tally'
  | 'times'
  | 'trash-alt'
  | 'triangle'
  | 'undo'
  | 'user-check'
  | 'user-group'
  | 'user-plus'
  | 'user-slash'
  | 'user'
  | 'users-slash'
  | 'venus'
>

const icons: { [key in IconName]: { light: IconDefinition; solid: IconDefinition } } = {
  'align-justify': { light: faAlignJustifyLight, solid: faAlignJustifySolid },
  analytics: { light: faAnalyticsLight, solid: faAnalyticsSolid },
  'angle-down': { light: faAngleDownLight, solid: faAngleDownSolid },
  'angle-up': { light: faAngleUpLight, solid: faAngleUpSolid },
  'arrow-circle-right': { light: faArrowCircleRightLight, solid: faArrowCircleRightSolid },
  'arrow-up-from-bracket': { light: faArrowUpFromBracketLight, solid: faArrowUpFromBracketSolid },
  'arrow-up-from-square': { light: faArrowUpFromSquareLight, solid: faArrowUpFromSquareSolid },
  'arrows-rotate': { light: faArrowsRotateLight, solid: faArrowsRotateSolid },
  'badge-check': { light: faBadgeCheckLight, solid: faBadgeCheckSolid },
  badge: { light: faBadgeLight, solid: faBadgeSolid },
  'ban-bug': { light: faBanBugLight, solid: faBanBugSolid },
  'calendar-day': { light: faCalendarDayLight, solid: faCalendarDaySolid },
  'calendar-range': { light: faCalendarRangeLight, solid: faCalendarRangeSolid },
  camera: { light: faCameraLight, solid: faCameraSolid },
  certificate: { light: faCertificateLight, solid: faCertificateSolid },
  'chart-column': { light: faChartColumnLight, solid: faChartColumnSolid },
  'check-circle': { light: faCheckCircleLight, solid: faCheckCircleSolid },
  'check-square': { light: faCheckSquareLight, solid: faCheckSquareSolid },
  check: { light: faCheckLight, solid: faCheckSolid },
  'chevron-circle-left': { light: faChevronCircleLeftLight, solid: faChevronCircleLeftSolid },
  'chevron-circle-right': { light: faChevronCircleRightLight, solid: faChevronCircleRightSolid },
  'chevron-left': { light: faChevronLeftLight, solid: faChevronLeftSolid },
  'chevron-right': { light: faChevronRightLight, solid: faChevronRightSolid },
  'circle-check': { light: faCircleCheckLight, solid: faCircleCheckSolid },
  'circle-exclamation': { light: faCircleExclamationLight, solid: faCircleExclamationSolid },
  'circle-plus': { light: faCirclePlusLight, solid: faCirclePlusSolid },
  'circle-stop': { light: faCircleStopLight, solid: faCircleStopSolid },
  'circle-user': { light: faCircleUserLight, solid: faCircleUserSolid },
  'circle-xmark': { light: faCircleXmarkLight, solid: faCircleXmarkSolid },
  circle: { light: faCircleLight, solid: faCircleSolid },
  clone: { light: faCloneLight, solid: faCloneSolid },
  'cloud-arrow-down': { light: faCloudArrowDownLight, solid: faCloudArrowDownSolid },
  cog: { light: faCogLight, solid: faCogSolid },
  'comment-alt-exclamation': { light: faCommentAltExclamationLight, solid: faCommentAltExclamationSolid },
  'crop-alt': { light: faCropAltLight, solid: faCropAltSolid },
  dash: { light: faDashLight, solid: faDashSolid },
  envelope: { light: faEnvelopeLight, solid: faEnvelopeSolid },
  'exclamation-triangle': { light: faExclamationTriangleLight, solid: faExclamationTriangleSolid },
  'external-link': { light: faExternalLinkLight, solid: faExternalLinkSolid },
  'eye-slash': { light: faEyeSlashLight, solid: faEyeSlashSolid },
  eye: { light: faEyeLight, solid: faEyeSolid },
  'file-lines': { light: faFileLinesLight, solid: faFileLinesSolid },
  'globe-europe': { light: faGlobeEuropeLight, solid: faGlobeEuropeSolid },
  heart: { light: faHeartLight, solid: faHeartSolid },
  hexagon: { light: faHexagonLight, solid: faHexagonSolid },
  images: { light: faImagesLight, solid: faImagesSolid },
  'info-circle': { light: faInfoCircleLight, solid: faInfoCircleSolid },
  language: { light: faLanguageLight, solid: faLanguageSolid },
  'location-check': { light: faLocationCheckLight, solid: faLocationCheckSolid },
  'location-crosshairs': { light: faLocationCrosshairsLight, solid: faLocationCrosshairsSolid },
  'map-marker-alt-slash': { light: faMapMarkerAltSlashLight, solid: faMapMarkerAltSlashSolid },
  'map-marker-alt': { light: faMapMarkerAltLight, solid: faMapMarkerAltSolid },
  mars: { light: faMarsLight, solid: faMarsSolid },
  minus: { light: faMinusLight, solid: faMinusSolid },
  'paper-plane-top': { light: faPaperPlaneToplight, solid: faPaperPlaneTopSolid },
  'pen-to-square': { light: faPenToSquareLight, solid: faPenToSquareSolid },
  'play-circle': { light: faPlayCircleLight, solid: faPlayCircleSolid },
  plus: { light: faPlusLight, solid: faPlusSolid },
  qrcode: { light: faQrcodeLight, solid: faQrcodeSolid },
  'question-square': { light: faQuestionSquareLight, solid: faQuestionSquareSolid },
  random: { light: faRandomLight, solid: faRandomSolid },
  'rectangle-history': { light: faRectangleHistoryLight, solid: faRectangleHistorySolid },
  'right-to-bracket': { light: faRightToBracketLight, solid: faRightToBracketSolid },
  route: { light: faRouteLight, solid: faRouteSolid },
  'share-nodes': { light: faShareNodesLight, solid: faShareNodesSolid },
  shield: { light: faShieldLight, solid: faShieldSolid },
  'shield-check': { light: faShieldCheckLight, solid: faShieldCheckSolid },
  'sliders-h': { light: faSlidersHLight, solid: faSlidersHSolid },
  square: { light: faSquareLight, solid: faSquareSolid },
  stars: { light: faStarsLight, solid: faStarsSolid },
  swatchbook: { light: faSwatchbookLight, solid: faSwatchbookSolid },
  tally: { light: faTallyLight, solid: faTallySolid },
  times: { light: faTimesLight, solid: faTimesSolid },
  'trash-alt': { light: faTrashAltLight, solid: faTrashAltSolid },
  triangle: { light: faTriangleLight, solid: faTriangleSolid },
  undo: { light: faUndoLight, solid: faUndoSolid },
  'user-check': { light: faUserCheckLight, solid: faUserCheckSolid },
  'user-group': { light: faUserGroupLight, solid: faUserGroupSolid },
  'user-plus': { light: faUserPlusLight, solid: faUserPlusSolid },
  'user-slash': { light: faUserSlashLight, solid: faUserSlashSolid },
  user: { light: faUserLight, solid: faUserSolid },
  'users-slash': { light: faUsersSlashLight, solid: faUsersSlashSolid },
  venus: { light: faVenusLight, solid: faVenusSolid },
}

export default icons

export type { IconName }
