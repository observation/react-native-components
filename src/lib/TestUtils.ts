const mockNavigation: any = {
  navigate: jest.fn(),
  addListener: jest.fn(() => jest.fn()),
  replace: jest.fn(),
  goBack: jest.fn(),
  popToTop: jest.fn(),
  canGoBack: jest.fn(),
  setOptions: jest.fn(),
  getState: jest.fn(() => ({ index: 0 })),
}

const mockNavigationContext: any = {
  isFocused: jest.fn(() => true),
  addListener: jest.fn(() => jest.fn()),
  navigate: jest.fn(),
}

export { mockNavigation, mockNavigationContext }
