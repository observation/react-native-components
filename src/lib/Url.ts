import { Linking } from 'react-native'

import Log from '../lib/Log'

export const openUrl = (url: string) =>
  Linking.canOpenURL(url).then((supported) => {
    if (supported) {
      return Linking.openURL(url)
    }
    Log.debug('Url:openUrl does not know how to open link', url)
  })
