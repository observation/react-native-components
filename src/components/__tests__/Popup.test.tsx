import React from 'react'
import { Text } from 'react-native'

import { NavigationContext } from '@react-navigation/native'
import { render } from '@testing-library/react-native'

import { mockNavigationContext } from '../../lib/TestUtils'
import Popup from '../Popup'

describe('Popup', () => {
  test('Rendering', () => {
    const { toJSON } = render(
      <NavigationContext.Provider value={mockNavigationContext}>
        <Popup visible={true}>
          <Text>Popup</Text>
        </Popup>
      </NavigationContext.Provider>,
    )
    expect(toJSON()).toMatchSnapshot()
  })
})
