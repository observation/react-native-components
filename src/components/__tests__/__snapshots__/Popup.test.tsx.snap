// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popup Rendering 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  style={
    {
      "flex": 1,
    }
  }
  transparent={true}
  visible={true}
>
  <BlurView
    blurAmount={2}
    blurType="light"
    style={
      [
        {
          "backgroundColor": "transparent",
        },
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
      ]
    }
  />
  <View
    style={
      {
        "alignItems": "stretch",
        "backgroundColor": "hsla(0, 0%, 18.1%, 0.6)",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "transparent",
          "flexDirection": "column",
          "justifyContent": "center",
        }
      }
    >
      <Text>
        Popup
      </Text>
    </View>
  </View>
</Modal>
`;
